lua/bufferline/layout.lua

```diff
@@ -68,6 +69,8 @@ function Layout.calculate_buffer_width(bufnr, index, diagnostics, use_buffer_ind
 
     if use_buffer_index then
       width = width + #tostring(index) + 1 -- buffer-index + space after buffer-index
+    elseif use_buffer_number then
```

Comment:
Turns out we forgot to account for buffer _numbers_ (not indices) in the layout code. I checked and this has been a bug since before 1.0.0.